graph.py

```diff
@@ -91,7 +95,7 @@ def depth_first_traversal(self, node,result=""):
         if not str(node) in result:
             result = '{}{}'.format(result, node)
```

Comment:
making the result of this a string is not a good choice.  It should be a list of the nodes visited in order.